### ISSUE TITLE
Your extension is affected by event stream and have been blocked

### ISSUE THREAD

**Issue by auchenberg** [Codeneos/vlocode#1]
Hi,

Kenneth here from the VS Code team. 

Your extension is affected by https://code.visualstudio.com/blogs/2018/11/26/event-stream, and we have blocked your extension.

In order to enable your extension again the workflow is:
  
1) Fix the extension, submit an update to the marketplace
2) Send mail to VSMarketplace@microsoft.com and kennau@microsoft.com and notify us.
3) We will verify that the extension is okay
4) Our marketplace will publish the extension again and we'll remove the extension from the blocklist

**Comment by Codeneos** [comment]
Thanks for the notification; looks like the event stream package is referenced by one of my dependencies. I will publish an update version tomorrow that addresses the issue.

**Comment by egamma** [comment]
@Codeneos you have 'vscode' listed as a dependency. Is there any reason for this? We recommend that 'vscode' is a devDependency. We would like to understand the reason why some extension authors change 'vscode' to be a 'dependency'.

**Comment by Codeneos** [comment]
@egamma I'm using webpack to package the extension, to get webpack to work correctly I need to pass `vscode` as external dependency to avoid having webpack trying to resolve it and include it in my final output bundle. 

To do this I parse my extension's package JSON in my webpack common config file and build the external module map from the dependencies defined in my package JSON:
```js
const packageJson = require("./package.json");
const packageExternals = Object.keys(packageJson.dependencies)
        .reduce((externals, dep) => Object.assign(externals, { [dep]: dep }), {});
```

I recently refactored that piece of code so that I don't have to list vscode as non-dev dependency anymore by using a separate list of externals.

In summary; 
1. I wasn't aware of any negative side effects that listing `vscode` as dependency would have.
2. In my specific case I am using webpack to package my extension which is a bit tricky adn as such I ended up listing it as dependency.

**Comment by Codeneos** [comment]
@auchenberg ; I have published an update (v0.8.6) that resolve the security issue by removing the dependency that relied on the corrupted version of `ps-tree` and referenced `flatstream-map`

**Comment by egamma** [comment]
@Codeneos we are also using webpack to bundle our extension, but we do not need to add vscode as dependency. Here is an example for how to use webpack to bundle an extension https://github.com/Microsoft/vscode-extension-samples/tree/master/webpack-sample

**Comment by Codeneos** [comment]
@egamma Thanks that is quite helpful! I referenced vscode as dependency so that I can iterate over the dependencies in webpack and build my `externals` webpack config property based on that. I refactored it in my last changeset to:

```
const externals = [
   // In order to run tests the main test frameworks need to be marked
   // as external to avoid conflicts when loaded by the VSCode test runner
   'mocha',
   'chai',
   'sinon',
   // VSCode is an external that we do not want to package
   'vscode', 
   'vscode-languageclient'
];

const packageJson = require("./package.json");
const packageExternals = Object.keys(packageJson.dependencies)
        .concat(externals)
        .reduce((externals, dep) => Object.assign(externals, { [dep]: dep }), {});
```

Though not stated in the example I also needed to reference `mocha` as external else unit testing would not work on the webpacked extension.
